specification/appplatform/resource-manager/Microsoft.AppPlatform/preview/2020-11-01-preview/appplatform.json

```diff
@@ -4014,6 +4018,20 @@
         }
       }
     },
+    "ResourceRequests": {
+      "description": "Deployment resource request payload",
+      "type": "object",
+      "properties": {
+        "cpu": {
```

Comment:
>  "cpu": { [](start = 6, length = 10)

As the property name this is not clear how the unit should be. I'd recommend making the name a bit clear and changing the type to integer. Please see an example of how both cpu and memory can be defined here: [https://github.com/Azure/azure-rest-api-specs/blob/abfe5f82e8ac29a05070957bc828874a625079f1/specification/compute/resource-manager/Microsoft.Compute/stable/2020-12-01/compute.json#L7609](https://github.com/Azure/azure-rest-api-specs/blob/abfe5f82e8ac29a05070957bc828874a625079f1/specification/compute/resource-manager/Microsoft.Compute/stable/2020-12-01/compute.json#L7609)

Reply:
It is by design, we want to make cpu and memory field in our API align with the AKS Resource requests convention: https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/

For cpu, the user can pass 1 or 1000m to represent the 1c.
For memory, the user can pass 1024Mi or 1Gi to represent the 1Gb.

For now, our system needs to support the size like 0.5c/0.5Gb App, the previous cpu and memoryInGb properties are integer type, so they can't represent the 0.5c/0.5Gb size, so we introduce a new field resourceRequests into the API request.

```
properties: {
  cpu: string
  memoryInGb: string
  resourceRequests: {
    cpu: string
    memory: string
  }
}
```
The user can use ethier cpu/memory or resourceRequests to represent his App's size.

Reply:
I see - thanks for the explanation.